src/test/java/com/epam/tc/hw3/testcases/Exercise1.java

```diff
@@ -0,0 +1,170 @@
+        }
+        softly.assertAll();
+    }
+
+    @Test(priority = 2)
+    public void assertIconsText() {
+        SoftAssertions softly = new SoftAssertions();
+        HomePage homePage = new HomePage(testDriver);
+
+        //7. Assert that there are 4 texts on the Index Page under icons and they have proper text
+        softly.assertThat(homePage.numberOfBenefitImagesText())
+                .as("Number of texts are " + homePage.numberOfBenefitImagesText() + " not 4")
+                .isEqualTo(4);
+
+        List<String> textUnderImagesExpected = Arrays.asList(
+                "To include good practices\nand ideas from successful\nEPAM project",
+                "To be flexible and\ncustomizable",
+                "To be multiplatform",
+                "Already have good base\n"
+                        + "(about 20 internal and\n"
+                        + "some external projects),\nwish to get more…"
+        );
+
+        List<String> textUnderImagesActual = homePage.getBenefitImagesText().stream().map(WebElement::getText)
+                .collect(Collectors.toList());
+        softly.assertThat(textUnderImagesActual).as("Text under icon differs").isEqualTo(textUnderImagesExpected);
+        softly.assertAll();
+    }
+
+    @Test(priority = 2)
+    public void assertFrames() {
+        SoftAssertions softly = new SoftAssertions();
+        HomePage homePage = new HomePage(testDriver);
+
+        //8. Assert that there is the iframe with “Frame Button” exist
+        int size = homePage.getIframes().size();
+        System.out.println(size);
+        int buttonInCurrentFrame = 0; //count buttons “Frame Button” in current frame
+        int totalButtonsInFrames = 0; //counter of frames with button “Frame Button”
+        for (int i = 0; i < size; i++) {
+
+            //9. Switch to the iframe and check that there is “Frame Button” in the iframe
+            testDriver.switchTo().frame(i);
+            buttonInCurrentFrame = homePage.getFrameButtons().size();
+
+            if (buttonInCurrentFrame > 0) {
+                totalButtonsInFrames = totalButtonsInFrames + buttonInCurrentFrame;
+                softly.assertThat(homePage.getCurrentFrameButton().getAttribute("value"))
+                        .as("Frame Button text differs")
+                        .isEqualTo("Frame Button");
+            }
+
+            System.out.println(buttonInCurrentFrame);
+
+            //10. Switch to original window back
+            testDriver.switchTo().defaultContent();
+        }
+        softly.assertThat(totalButtonsInFrames).as("Button not found").isNotZero();
+
+        softly.assertAll();
+    }
+
+    @Test(priority = 2)
+    public void assertLeftMenu() {
+        SoftAssertions softly = new SoftAssertions();
+
+        //11. Assert that there are 5 items in the Left Section are displayed and they have proper text
+        SideBar sideBar = new SideBar(testDriver);
```

Comment:
Please read this very good article https://habr.com/ru/company/jugru/blog/338836/

Make this code according to Page Element pattern